assertk/src/commonMain/kotlin/assertk/assertions/iterable.kt

```diff
@@ -131,6 +133,34 @@ internal fun MutableList<*>.removeFirst(value: Any?) {
     if (index > -1) removeAt(index)
 }
 
+/**
+ * Asserts the collection contains at least one instance of a given type.
+ *
+ * ```
+ * assertThat(listOf<Any>("one", "two", 1)).containsInstanceOf<String>().each {
+ *   it.hasLength(3)
+ * }
+ * ```
+ */
+inline fun <reified T> Assert<Iterable<*>>.containsInstanceOf(): Assert<List<T>> {
+    return transform("contains subtype of ${T::class}") { actual ->
```

Comment:
This message should probably just be `contains instance of ${T::class}` per Liskov.